components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/builder/WSO2LoginEventPayloadBuilder.java

```diff
@@ -17,53 +17,207 @@
+        }
+
+        User user = new User();
+        try {
+            populateUserAttributes(authenticatedUser, user);
+            user.setId(authenticatedUser.getUserId());
+            user.setRef(EventHookHandlerUtils.getInstance().getURL(Constants.SCIM2_ENDPOINT) + "/" +
+                    authenticatedUser.getUserId());
+        } catch (UserIdNotFoundException e) {
+            throw new IdentityEventException("Error while building the event payload", e);
+        }
+
+        Organization tenant = new Organization(
+                String.valueOf(IdentityTenantUtil.getTenantId(authenticationContext.getTenantDomain())),
+                authenticationContext.getTenantDomain());
+        UserStore userStore = null;
+        if (authenticatedUser.getUserStoreDomain() != null) {
+            userStore = new UserStore(authenticatedUser.getUserStoreDomain());
+        }
+        Organization b2bUserResidentOrganization = null;
+        if (authenticatedUser.getUserResidentOrganization() != null) {
+            b2bUserResidentOrganization = getUserResidentOrganization(
+                    authenticatedUser.getUserResidentOrganization());
+        }
+        Application application = new Application(
+                authenticationContext.getServiceProviderResourceId(),
+                authenticationContext.getServiceProviderName());
         return new WSO2AuthenticationSuccessEventPayload.Builder()
-                .user(null)
-                .tenant(null)
-                .userResidentOrganization(null)
-                .userStore(null)
-                .application(null)
-                .authenticationMethods(new ArrayList<>())
+                .user(user)
+                .tenant(tenant)
+                .userResidentOrganization(b2bUserResidentOrganization)
+                .userStore(userStore)
+                .application(application)
+                .authenticationMethods(buildAuthMethods(authenticationContext))
                 .build();
     }
 
     @Override
     public EventPayload buildAuthenticationFailedEvent(EventData eventData) throws IdentityEventException {
 
-        //TODO: Implement the logic to build the authentication failed event payload.
+        AuthenticationContext authenticationContext = eventData.getAuthenticationContext();
+        AuthenticatedUser authenticatedUser = authenticationContext.getSubject();
+
+        User user = new User();
+        UserStore userStore = null;
+        try {
+            if (authenticatedUser != null) {
+                user.setId(authenticatedUser.getUserId());
+                user.setRef(EventHookHandlerUtils.getInstance().getURL(Constants.SCIM2_ENDPOINT) + "/" +
+                        authenticatedUser.getUserId());
+                if (authenticatedUser.getUserStoreDomain() != null) {
+                    userStore = new UserStore(authenticatedUser.getUserStoreDomain());
+                }
```

Comment:
Take setting user store domain out of this block. It won't be set of user id is not found in this case

Reply:
Addressed with https://github.com/Sachin-Mamoru/identity-webhook-event-handlers/pull/1/commits/8750eff9ee3a84ff76fd11671bbbc3451fbb0fc8